app/src/modelProviders/fine-tuned/getCompletion-2.ts

```diff
@@ -21,7 +23,7 @@ import { type TypedFineTune } from "~/types/dbColumns.types";
 export async function getCompletion2(
```

Comment:
So I realize it looks ugly but maybe we want to add multiple function declarations here? That way typescript knows that if we pass in stream: true, we get a stream, and if not not. Eg:

export async function getCompletion2(
  fineTune: TypedFineTune,
  input: ChatCompletionCreateParamsNonStreaming,
): Promise<ChatCompletion>;
export async function getCompletion2(
  fineTune: TypedFineTune,
  input: ChatCompletionCreateParamsStreaming,
): Promise<Stream<ChatCompletionChunk>>;
export async function getCompletion2(
  fineTune: TypedFineTune,
  input: ChatCompletionCreateParams,
): Promise<ChatCompletion | Stream<ChatCompletionChunk>>;

Reply:
It's not ugly. It's beautiful.